### PR TITLE
Implement MSLE (MeanSquaredLogarithmicError) evaluation metric for regression analysis

### LINKED ISSUE #58682: Implement MSLE function for regression evaluation.
Regression evaluation should have an option to use MSLE function.

https://peltarion.com/knowledge-center/documentation/modeling-view/build-an-ai-model/loss-functions/mean-squared-logarithmic-error-(msle)

### PR DESCRIPTION
**Pull request by przemekwitek**
This PR implements MSLE evaluation metric for regression analysis.

Additionally, this PR adds `RegressionEvaluationIT` integration tests class. 

Closes https://github.com/elastic/elasticsearch/issues/58682

### CODE REVIEW

**Comment by przemekwitek** [comment]
run elasticsearch-ci/bwc

File: client/rest-high-level/src/main/java/org/elasticsearch/client/ml/dataframe/evaluation/regression/MeanSquaredLogarithmicErrorMetric.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        }
+
+        public Result(double error) {
+            this.error = error;
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, ToXContent.Params params) throws IOException {
+            builder.startObject();
+            builder.field(ERROR.getPreferredName(), error);
```
Comment (davidkyle) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/ml/dataframe/evaluation/regression/MeanSquaredLogarithmicErrorMetric.java:115]: The field name "error" is confusing could this be "msle"

File: x-pack/plugin/ml/qa/native-multi-node-tests/src/test/java/org/elasticsearch/xpack/ml/integration/RegressionEvaluationIT.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+        createHousesIndex(HOUSES_DATA_INDEX);
+        indexHousesData(HOUSES_DATA_INDEX);
+    }
+
+    @After
+    public void cleanup() {
+        cleanUp();
+        client().admin().cluster()
+            .prepareUpdateSettings()
+            .setTransientSettings(Settings.builder().putNull("search.max_buckets"))
```
Comment (davidkyle) [inline x-pack/plugin/ml/qa/native-multi-node-tests/src/test/java/org/elasticsearch/xpack/ml/integration/RegressionEvaluationIT.java:67]: ?? I would expect this to be set in the test setup but I don't see that

File: client/rest-high-level/src/main/java/org/elasticsearch/client/ml/dataframe/evaluation/regression/MeanSquaredLogarithmicErrorMetric.java (hunk 1 above)
Comment (davidkyle) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/ml/dataframe/evaluation/regression/MeanSquaredLogarithmicErrorMetric.java:115]: Actually I see MeanSquaredError also has a field "error" and in the context of the result object it makes sense

**Review by davidkyle — APPROVED** [review]
LGTM

**Comment by davidkyle** [comment]
The bwc failure is due to #58722 (I hit the same problem)

**Comment by przemekwitek** [comment]
run elasticsearch-ci/bwc

**Comment by przemekwitek** [comment]
run elasticsearch-ci/bwc

**Pull request merged**